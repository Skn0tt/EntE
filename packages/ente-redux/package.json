{
  "name": "ente-redux",
  "version": "0.2.3",
  "description": "State Management for EntE",
  "main": "src/index.ts",
  "repository": "gitlab.com/skn0tt/EntE/master/packages/ente-redux",
  "license": "AGPL-3.0-or-later",
  "author": {
    "email": "info@simonknott.de",
    "name": "Simon Knott",
    "url": "https://simonknott.de"
  },
  "private": true,
  "dependencies": {
    "axios": "^0.18.0",
    "ente-lang": "^0.2.3",
    "ente-types": "^0.2.3",
    "immutable": "^3.8.2",
    "redux": "^3.7.2",
    "redux-actions": "^2.3.0",
    "redux-saga": "^0.16.0",
    "reselect": "^3.0.1"
  },
  "devDependencies": {
    "@types/redux-actions": "^2.2.4"
  }
}
