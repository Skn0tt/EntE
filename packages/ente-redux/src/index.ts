/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import { createStore } from "./store";
import { Middleware } from "redux";

export * from "./selectors";
export * from "./actions";
export * from "./types";
export * from "./constants";

export type ReduxConfig = {
  baseUrl: string;
  middlewares?: Middleware[];
  onSagaError?: (err: Error) => void;
  onFileDownload: (file: Blob, filename: string) => void;
};

export let config: ReduxConfig = {
  baseUrl: "",
  middlewares: [],
  onFileDownload: () => {}
};

const setup = (conf: ReduxConfig) => {
  config = conf;

  return createStore();
};

export default setup;
