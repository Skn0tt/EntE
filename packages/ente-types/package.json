{
  "name": "ente-types",
  "version": "0.15.0",
  "description": "Typings for EntE",
  "main": "src/index.ts",
  "repository": "gitlab.com/skn0tt/EntE/master/ente-types",
  "license": "AGPL-3.0-or-later",
  "author": {
    "email": "info@simonknott.de",
    "name": "Simon Knott",
    "url": "https://simonknott.de"
  },
  "private": true,
  "scripts": {
    "build": "yarn tsc"
  },
  "devDependencies": {
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "class-transformer": "0.1.10",
    "class-validator": "^0.9.1",
    "date-fns": "^2.0.0-alpha.26"
  }
}
