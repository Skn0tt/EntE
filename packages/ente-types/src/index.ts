/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

export * from "./roles";
export * from "./dtos";
export * from "./validators";
export * from "./helpers";
