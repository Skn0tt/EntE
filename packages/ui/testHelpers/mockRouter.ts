import { RouteComponentProps } from "react-router";
import { Location, UnregisterCallback } from "history";

export function getMockRouterProps<P>(data: P) {
  const location: Location = {
    hash: "",
    key: "",
    pathname: "",
    search: "",
    state: {}
  };

  const props: RouteComponentProps<P> = {
    location,
    match: {
      isExact: true,
      params: data,
      path: "",
      url: ""
    },
    history: {
      location,
      length: 2,
      action: "POP",
      push: () => {},
      replace: () => {},
      go: num => {},
      goBack: () => {},
      goForward: () => {},
      block: t => null as any,
      createHref: t => "",
      listen: t => null as any
    },
    staticContext: {}
  };

  return props;
}
