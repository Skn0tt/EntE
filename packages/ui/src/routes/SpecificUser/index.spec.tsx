import * as React from "react";
import { SpecificUser } from "./index";
import { shallow } from "enzyme";
import { User } from "ente-redux";
import { getMockRouterProps } from "../../../testHelpers/mockRouter";
import { Roles } from "ente-types";
import * as sinon from "sinon";

describe("SpecificUser", () => {
  const userId = "fdas90ß9sß0";
  const user = new User({
    _id: userId,
    children: [],
    displayname: "Horst Hansen",
    email: "hort@hansen.de",
    isAdult: false,
    role: Roles.STUDENT,
    username: "hhansen"
  });
  const getUser = sinon.stub().returns(user);
  const requestUser = sinon.spy();

  const comp = shallow(
    <SpecificUser
      {...getMockRouterProps({ userId })}
      getUser={getUser}
      fullScreen
      classes={{}}
      loading={false}
      requestUser={requestUser}
      students={[]}
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  it("requests the user", () => {
    expect(getUser.calledWith(userId)).toBe(true);
  });

  describe("when `getUser` doesn't return a user", () => {
    const getUser = sinon.stub().returns(undefined);
    const requestUser = sinon.spy();
    const comp = shallow(
      <SpecificUser
        {...getMockRouterProps({ userId })}
        getUser={getUser}
        fullScreen
        classes={{}}
        loading={false}
        requestUser={requestUser}
        students={[]}
      />
    );

    it("requests the user", () => {
      expect(requestUser.called).toBe(true);
    });
  });
});
