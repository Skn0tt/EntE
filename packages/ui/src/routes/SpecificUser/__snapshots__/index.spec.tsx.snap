// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpecificUser renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpecificUser
    classes={Object {}}
    fullScreen={true}
    getUser={[Function]}
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": Object {},
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    loading={false}
    location={
      Object {
        "hash": "",
        "key": "",
        "pathname": "",
        "search": "",
        "state": Object {},
      }
    }
    match={
      Object {
        "isExact": true,
        "params": Object {
          "userId": "fdas90ß9sß0",
        },
        "path": "",
        "url": "",
      }
    }
    requestUser={[Function]}
    staticContext={Object {}}
    students={Array []}
    updateUser={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <UNDEFINED>
          <WithStyles(DialogTitle)>
            Horst Hansen
          </WithStyles(DialogTitle)>
          <WithStyles(DialogContent)>
            <WithStyles(GridWrapper)
              alignItems="stretch"
              container={true}
              spacing={24}
            >
              <WithStyles(GridWrapper)
                item={true}
                xs={12}
              >
                <WithStyles(DialogContentText)>
                  ID
                  : 
                  fdas90ß9sß0
                   
                  <br />
                  Rolle
                  : 
                  student
                   
                  <br />
                </WithStyles(DialogContentText)>
              </WithStyles(GridWrapper)>
              <WithStyles(GridWrapper)
                item={true}
                xs={12}
              >
                <Unknown
                  onChange={[Function]}
                  title="Displayname"
                  validator={[Function]}
                  value="Horst Hansen"
                />
              </WithStyles(GridWrapper)>
              <WithStyles(GridWrapper)
                item={true}
                xs={12}
              >
                <Unknown
                  onChange={[Function]}
                  title="Email"
                  validator={[Function]}
                  value="hort@hansen.de"
                />
              </WithStyles(GridWrapper)>
              <WithStyles(GridWrapper)
                item={true}
                xs={12}
              >
                <Unknown
                  onChange={[Function]}
                  title="Erwachsen"
                  value={false}
                />
              </WithStyles(GridWrapper)>
            </WithStyles(GridWrapper)>
          </WithStyles(DialogContent)>
        </UNDEFINED>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="secondary"
            onClick={[Function]}
            size="small"
          >
            Schließen
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            disabled={false}
            onClick={[Function]}
            size="small"
          >
            OK
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "fullScreen": true,
      "onClose": [Function],
      "open": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <WithStyles(DialogTitle)>
              Horst Hansen
            </WithStyles(DialogTitle)>,
            <WithStyles(DialogContent)>
              <WithStyles(GridWrapper)
                alignItems="stretch"
                container={true}
                spacing={24}
              >
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <WithStyles(DialogContentText)>
                    ID
                    : 
                    fdas90ß9sß0
                     
                    <br />
                    Rolle
                    : 
                    student
                     
                    <br />
                  </WithStyles(DialogContentText)>
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Displayname"
                    validator={[Function]}
                    value="Horst Hansen"
                  />
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Email"
                    validator={[Function]}
                    value="hort@hansen.de"
                  />
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Erwachsen"
                    value={false}
                  />
                </WithStyles(GridWrapper)>
              </WithStyles(GridWrapper)>
            </WithStyles(DialogContent)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Horst Hansen",
            },
            "ref": null,
            "rendered": "Horst Hansen",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(GridWrapper)
                alignItems="stretch"
                container={true}
                spacing={24}
              >
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <WithStyles(DialogContentText)>
                    ID
                    : 
                    fdas90ß9sß0
                     
                    <br />
                    Rolle
                    : 
                    student
                     
                    <br />
                  </WithStyles(DialogContentText)>
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Displayname"
                    validator={[Function]}
                    value="Horst Hansen"
                  />
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Email"
                    validator={[Function]}
                    value="hort@hansen.de"
                  />
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Erwachsen"
                    value={false}
                  />
                </WithStyles(GridWrapper)>
              </WithStyles(GridWrapper)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alignItems": "stretch",
                "children": Array [
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <WithStyles(DialogContentText)>
                      ID
                      : 
                      fdas90ß9sß0
                       
                      <br />
                      Rolle
                      : 
                      student
                       
                      <br />
                    </WithStyles(DialogContentText)>
                  </WithStyles(GridWrapper)>,
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Displayname"
                      validator={[Function]}
                      value="Horst Hansen"
                    />
                  </WithStyles(GridWrapper)>,
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Email"
                      validator={[Function]}
                      value="hort@hansen.de"
                    />
                  </WithStyles(GridWrapper)>,
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Erwachsen"
                      value={false}
                    />
                  </WithStyles(GridWrapper)>,
                  false,
                ],
                "container": true,
                "spacing": 24,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(DialogContentText)>
                      ID
                      : 
                      fdas90ß9sß0
                       
                      <br />
                      Rolle
                      : 
                      student
                       
                      <br />
                    </WithStyles(DialogContentText)>,
                    "item": true,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "ID",
                        ": ",
                        "fdas90ß9sß0",
                        " ",
                        <br />,
                        "Rolle",
                        ": ",
                        "student",
                        " ",
                        <br />,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "ID",
                      ": ",
                      "fdas90ß9sß0",
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                      "Rolle",
                      ": ",
                      "student",
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "br",
                      },
                    ],
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      onChange={[Function]}
                      title="Displayname"
                      validator={[Function]}
                      value="Horst Hansen"
                    />,
                    "item": true,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "onChange": [Function],
                      "title": "Displayname",
                      "validator": [Function],
                      "value": "Horst Hansen",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      onChange={[Function]}
                      title="Email"
                      validator={[Function]}
                      value="hort@hansen.de"
                    />,
                    "item": true,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "onChange": [Function],
                      "title": "Email",
                      "validator": [Function],
                      "value": "hort@hansen.de",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Unknown
                      onChange={[Function]}
                      title="Erwachsen"
                      value={false}
                    />,
                    "item": true,
                    "xs": 12,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "onChange": [Function],
                      "title": "Erwachsen",
                      "value": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                false,
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
              size="small"
            >
              Schließen
            </WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              disabled={false}
              onClick={[Function]}
              size="small"
            >
              OK
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Schließen",
              "color": "secondary",
              "onClick": [Function],
              "size": "small",
            },
            "ref": null,
            "rendered": "Schließen",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "OK",
              "color": "primary",
              "disabled": false,
              "onClick": [Function],
              "size": "small",
            },
            "ref": null,
            "rendered": "OK",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <UNDEFINED>
            <WithStyles(DialogTitle)>
              Horst Hansen
            </WithStyles(DialogTitle)>
            <WithStyles(DialogContent)>
              <WithStyles(GridWrapper)
                alignItems="stretch"
                container={true}
                spacing={24}
              >
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <WithStyles(DialogContentText)>
                    ID
                    : 
                    fdas90ß9sß0
                     
                    <br />
                    Rolle
                    : 
                    student
                     
                    <br />
                  </WithStyles(DialogContentText)>
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Displayname"
                    validator={[Function]}
                    value="Horst Hansen"
                  />
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Email"
                    validator={[Function]}
                    value="hort@hansen.de"
                  />
                </WithStyles(GridWrapper)>
                <WithStyles(GridWrapper)
                  item={true}
                  xs={12}
                >
                  <Unknown
                    onChange={[Function]}
                    title="Erwachsen"
                    value={false}
                  />
                </WithStyles(GridWrapper)>
              </WithStyles(GridWrapper)>
            </WithStyles(DialogContent)>
          </UNDEFINED>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="secondary"
              onClick={[Function]}
              size="small"
            >
              Schließen
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              disabled={false}
              onClick={[Function]}
              size="small"
            >
              OK
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "fullScreen": true,
        "onClose": [Function],
        "open": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <WithStyles(DialogTitle)>
                Horst Hansen
              </WithStyles(DialogTitle)>,
              <WithStyles(DialogContent)>
                <WithStyles(GridWrapper)
                  alignItems="stretch"
                  container={true}
                  spacing={24}
                >
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <WithStyles(DialogContentText)>
                      ID
                      : 
                      fdas90ß9sß0
                       
                      <br />
                      Rolle
                      : 
                      student
                       
                      <br />
                    </WithStyles(DialogContentText)>
                  </WithStyles(GridWrapper)>
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Displayname"
                      validator={[Function]}
                      value="Horst Hansen"
                    />
                  </WithStyles(GridWrapper)>
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Email"
                      validator={[Function]}
                      value="hort@hansen.de"
                    />
                  </WithStyles(GridWrapper)>
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Erwachsen"
                      value={false}
                    />
                  </WithStyles(GridWrapper)>
                </WithStyles(GridWrapper)>
              </WithStyles(DialogContent)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Horst Hansen",
              },
              "ref": null,
              "rendered": "Horst Hansen",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(GridWrapper)
                  alignItems="stretch"
                  container={true}
                  spacing={24}
                >
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <WithStyles(DialogContentText)>
                      ID
                      : 
                      fdas90ß9sß0
                       
                      <br />
                      Rolle
                      : 
                      student
                       
                      <br />
                    </WithStyles(DialogContentText)>
                  </WithStyles(GridWrapper)>
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Displayname"
                      validator={[Function]}
                      value="Horst Hansen"
                    />
                  </WithStyles(GridWrapper)>
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Email"
                      validator={[Function]}
                      value="hort@hansen.de"
                    />
                  </WithStyles(GridWrapper)>
                  <WithStyles(GridWrapper)
                    item={true}
                    xs={12}
                  >
                    <Unknown
                      onChange={[Function]}
                      title="Erwachsen"
                      value={false}
                    />
                  </WithStyles(GridWrapper)>
                </WithStyles(GridWrapper)>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alignItems": "stretch",
                  "children": Array [
                    <WithStyles(GridWrapper)
                      item={true}
                      xs={12}
                    >
                      <WithStyles(DialogContentText)>
                        ID
                        : 
                        fdas90ß9sß0
                         
                        <br />
                        Rolle
                        : 
                        student
                         
                        <br />
                      </WithStyles(DialogContentText)>
                    </WithStyles(GridWrapper)>,
                    <WithStyles(GridWrapper)
                      item={true}
                      xs={12}
                    >
                      <Unknown
                        onChange={[Function]}
                        title="Displayname"
                        validator={[Function]}
                        value="Horst Hansen"
                      />
                    </WithStyles(GridWrapper)>,
                    <WithStyles(GridWrapper)
                      item={true}
                      xs={12}
                    >
                      <Unknown
                        onChange={[Function]}
                        title="Email"
                        validator={[Function]}
                        value="hort@hansen.de"
                      />
                    </WithStyles(GridWrapper)>,
                    <WithStyles(GridWrapper)
                      item={true}
                      xs={12}
                    >
                      <Unknown
                        onChange={[Function]}
                        title="Erwachsen"
                        value={false}
                      />
                    </WithStyles(GridWrapper)>,
                    false,
                  ],
                  "container": true,
                  "spacing": 24,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(DialogContentText)>
                        ID
                        : 
                        fdas90ß9sß0
                         
                        <br />
                        Rolle
                        : 
                        student
                         
                        <br />
                      </WithStyles(DialogContentText)>,
                      "item": true,
                      "xs": 12,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": Array [
                          "ID",
                          ": ",
                          "fdas90ß9sß0",
                          " ",
                          <br />,
                          "Rolle",
                          ": ",
                          "student",
                          " ",
                          <br />,
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        "ID",
                        ": ",
                        "fdas90ß9sß0",
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": "br",
                        },
                        "Rolle",
                        ": ",
                        "student",
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": "br",
                        },
                      ],
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Unknown
                        onChange={[Function]}
                        title="Displayname"
                        validator={[Function]}
                        value="Horst Hansen"
                      />,
                      "item": true,
                      "xs": 12,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "onChange": [Function],
                        "title": "Displayname",
                        "validator": [Function],
                        "value": "Horst Hansen",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Unknown
                        onChange={[Function]}
                        title="Email"
                        validator={[Function]}
                        value="hort@hansen.de"
                      />,
                      "item": true,
                      "xs": 12,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "onChange": [Function],
                        "title": "Email",
                        "validator": [Function],
                        "value": "hort@hansen.de",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Unknown
                        onChange={[Function]}
                        title="Erwachsen"
                        value={false}
                      />,
                      "item": true,
                      "xs": 12,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "onChange": [Function],
                        "title": "Erwachsen",
                        "value": false,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  false,
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="secondary"
                onClick={[Function]}
                size="small"
              >
                Schließen
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                disabled={false}
                onClick={[Function]}
                size="small"
              >
                OK
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Schließen",
                "color": "secondary",
                "onClick": [Function],
                "size": "small",
              },
              "ref": null,
              "rendered": "Schließen",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "OK",
                "color": "primary",
                "disabled": false,
                "onClick": [Function],
                "size": "small",
              },
              "ref": null,
              "rendered": "OK",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
